Code/Tools/AssetProcessor/native/utilities/PlatformConfiguration.cpp

```diff
@@ -1873,6 +1934,9 @@ namespace AssetProcessor
             {
                 const AZ::IO::Path& absoluteSourcePath = gemElement.m_absoluteSourcePaths[sourcePathIndex];
                 QString gemAbsolutePath = QString::fromUtf8(absoluteSourcePath.c_str(), aznumeric_cast<int>(absoluteSourcePath.Native().size())); // this is an absolute path!
+
+                bool isProjectGem = AZStd::find(projectGemPaths.begin(), projectGemPaths.end(), absoluteSourcePath) != projectGemPaths.end();
```

Comment:
You can use `AZStd::ranges::contain` here.
```suggestion
                const bool isProjectGem = AZStd::ranges::contain(projectGemPaths, absoluteSourcePath);
```

Reply:
Now that I think about this.
Instead of gathering the list of Project Gems, we can just check if the gem path is relative to the project root.

```suggestion
                const bool isProjectGem = gemAbsolutePath.IsRelativeTo(projectPath);
```